src/wms/src/main/java/org/geoserver/wms/GetMap.java

```diff
@@ -669,7 +669,24 @@ WebMap executeInternal(
                     }
                 }
                 if (!merged) {
-                    WMSLayer Layer = new WMSLayer(wms, gt2Layer);
+                    WMSLayer Layer = null;
+                    // GEOS-9312
+                    if (mapLayerInfo.getLayerInfo() instanceof CascadedLayerInfo) {
+                        CascadedLayerInfo cascadedLayerInfo =
+                                (CascadedLayerInfo) mapLayerInfo.getLayerInfo();
+
+                        String style = request.getStyles().get(i).getName();
+                        String imageFormat = request.getFormat();
+                        // if passed style does not exist in remote, default to forced
+                        if (!cascadedLayerInfo.isSelectedRemoteStyles(style))
+                            style = cascadedLayerInfo.getForcedRemoteStyle();
+
+                        // if the format is not selected then fall back to preffered
+                        if (!cascadedLayerInfo.isSelectedRemoteFormat(imageFormat))
```

Comment:
This doesn't make much sense to me, if the client requests a PNG and it is not supported by the remote, then it should query the remote using whatever is the default format, but return a PNG to the client.

Another thing, that behavior should be handle by the WMS remote store, not by GeoServer, although I'm not sure at 100% this is possible.

Reply:
the list comes from Capability document read during creation of store, even if its possible to re-query remote WMS service that will generate a lot of remote requests, No?

I will throw exception if the format is unknown

Reply:
What do you mean by this: "I will throw exception if the format is unknown"?

`the list comes from Capability document read during creation of store, even if its possible to re-query remote WMS service that will generate a lot of remote requests, No?`

No you shoudl not query it each time.

Reply:
understood